tutorials/yolov8_pose_e2e.py

```diff
@@ -0,0 +1,261 @@
+# Copyright (c) Microsoft Corporation.
+# Licensed under the MIT License.
+
+import onnx.shape_inference
+import onnxruntime_extensions
+from onnxruntime_extensions.tools.pre_post_processing import *
+from pathlib import Path
+from PIL import Image, ImageDraw
+
+def get_yolov8_pose_model(onnx_model_name: str):
+    # install yolov8
+    from pip._internal import main as pipmain
+    try:
+        import ultralytics
+    except ImportError:
+        pipmain(['install', 'ultralytics'])
+        import ultralytics
+    pt_model = Path("yolov8n-pose.pt")
+    model = ultralytics.YOLO(str(pt_model))  # load a pretrained model
+    success = model.export(format="onnx")  # export the model to ONNX format
+    assert success, "Failed to export yolov8n-pose.pt to onnx"
+    import shutil
+    shutil.move(pt_model.with_suffix('.onnx'), onnx_model_name)
+
+
+def add_pre_post_processing_to_yolo(input_model_file: Path, output_model_file: Path,
+                                    output_image: bool = False,
+                                    decode_input: bool = True,
+                                    input_shape: Optional[List[int | str]] = None):
```

Comment:
the expression `int | str` requires python3.10, right?

Reply:
Is there an alternative way that works with earlier versions other than `Any`? 

As this is model editing that can happen offline in any environment, requiring python 3.10 shouldn't be a blocker although it might be inconvenient.

Reply:
Could we use `Any` for python below 3.10 and `int | str` for 3.10+?
```
if sys.version_info.minor<10:
       shape_type_constrain = Any
else:
       shape_type_constrain = int | str
```
Then
```
input_shape: Optional[List[shape_type_constrain ]] = None):
```

Reply:
I'll go with this recommendation of using `Union` to keep things simple:

https://mypy.readthedocs.io/en/stable/cheat_sheet_py3.html

```py
# On Python 3.10+, use the | operator when something could be one of a few types
x: list[int | str] = [3, 5, "test", "fun"]  # Python 3.10+
# On earlier versions, use Union
x: list[Union[int, str]] = [3, 5, "test", "fun"]
```